### PR TITLE
fix: fixed `shape#array` types

### PR DESCRIPTION
**Pull request by imranbarbhuiya**

### CODE REVIEW

**Review by vladfrangu — DISMISSED** [review]

File: src/lib/Shapes.ts [hunk 1]
```diff
@@ -94,7 +94,7 @@ export class Shapes {
 		return new UnionValidator(validators);
 	}
 
-	public array<T>(validator: BaseValidator<T>) {
+	public array<T extends unknown[]>(validator: BaseValidator<T[number]>) {
```
Comment (kyranet) [inline src/lib/Shapes.ts:97]: What does this even fix? Can we please get unit tests?

**Review by kyranet — CHANGES_REQUESTED** [review]

File: src/lib/Shapes.ts (hunk 1 above)
Comment (imranbarbhuiya) [reply src/lib/Shapes.ts:97]: https://github.com/discordjs/discord.js/pull/8242

Comment (iCrawl) [reply src/lib/Shapes.ts:97]: Should probably still add a test case if you can

**Review by favna — APPROVED** [review]

**Comment by favna** [comment]
@sapphiredev pack

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**